vscode/vscode/vscode.mli

```diff
@@ -2118,26 +1913,36 @@ module Window : sig
     -> Terminal.t
 
   val registerTreeDataProvider :
-    viewId:string -> treeDataProvider:TreeDataProvider.t -> Disposable.t
+       (module Js.T with type t = 'a)
+    -> viewId:string
+    -> treeDataProvider:'a TreeDataProvider.t
+    -> Disposable.t
 
-  val createTreeView : viewId:string -> options:TreeViewOptions.t -> TreeView.t
+  val createTreeView :
+       (module Js.T with type t = 'a)
+    -> viewId:string
+    -> options:'a TreeViewOptions.t
+    -> 'a TreeView.t
 end
 
 module Commands : sig
   val registerCommand :
-    command:string -> callback:(args:Ojs.t list -> unit) -> Disposable.t
+    command:string -> callback:(args:Js.Any.t list -> unit) -> Disposable.t
 
   val registerTextEditorCommand :
        command:string
     -> callback:
          (   textEditor:TextEditor.t
           -> edit:TextEditorEdit.t
-          -> args:Ojs.t list
+          -> args:Js.Any.t list
           -> unit)
     -> Disposable.t
 
   val executeCommand :
```

Comment:
I don't think you should use pass a module in situations where it doesn't add any type safety. In this situation, returning `Ojs.t` seems to be just as good, and simpler to boot.